docs/shared/features/maintain-typescript-monorepos.md

```diff
@@ -0,0 +1,286 @@
+# Maintain TypeScript Monorepos
+
+Keeping all the industry-standard tools involved in a large TypeScript monorepo correctly configured and working well together is a difficult task. And the more tools you add, the more opportunity there is for tools to conflict with each other in some way.
+
+In addition to [generating default configuration files](/features/generate-code) and [automatically updating dependencies](/features/automate-updating-dependencies) to versions that we know work together, Nx makes managing all the tools in your monorepo easier in two ways:
+
+- Rather than adding another tool that you have to configure, Nx configures itself to match the existing configuration of other tools.
+- Nx also enhances certain tools to be more usable in a monorepo context.
+
+## Auto-Configuration
+
+Whenever possible, Nx will detect the existing configuration settings of other tools and update itself to match.
+
+### Project Detection with Workspaces
+
+If your repository is using package manager workspaces, Nx will use those settings to find all the projects in your repository. So you don't need to define a project for your package manager and separately identify the project for Nx. The `workspaces` configuration on the left, allows Nx to detect the project graph on the right.
+
+{% side-by-side align="top" %}
+
+```json {% fileName="/package.json" %}
+{
+  "workspaces": ["apps/**", "packages/**"]
+}
+```
+
+{% graph height="200px" title="Project View" %}
+
+```json
+{
+  "composite": false,
+  "projects": [
+    {
+      "name": "product-state",
+      "type": "lib",
+      "data": {
+        "root": "packages/cart/product-state",
+        "tags": ["scope:cart", "type:state"]
+      }
+    },
+    {
+      "name": "ui-buttons",
+      "type": "lib",
+      "data": {
+        "root": "packages/ui/buttons",
+        "tags": ["scope:shared", "type:ui"]
+      }
+    },
+    {
+      "name": "cart",
+      "type": "app",
+      "data": {
+        "root": "apps/cart",
+        "tags": ["type:app", "scope:cart"]
+      }
+    }
+  ],
+  "dependencies": {
+    "product-state": [],
+    "ui-buttons": [],
+    "cart": [
+      { "source": "cart", "target": "product-state", "type": "static" },
+      { "source": "cart", "target": "ui-buttons", "type": "static" }
+    ]
+  },
+  "workspaceLayout": {
+    "appsDir": "apps",
+    "libsDir": "libs"
+  },
+  "affectedProjectIds": [],
+  "focus": null,
+  "groupByFolder": false,
+  "exclude": [],
+  "enableTooltips": false
+}
+```
+
+{% /graph %}
+
+{% /side-by-side %}
+
+- Nx configures itself to match the existing configuration of other tools.
```

Comment:
This feels oddly places. Should this go under the main `Auto-Configuration` section, above `Project Detection with Workspaces`? Also, the ordering should match content: workspaces, inferred tasks, sync generators + project references.